File: app/src/main/java/com/rokudo/xpense/utils/DatabaseUtils.java
package com.rokudo.xpense.utils;

import com.google.firebase.firestore.CollectionReference;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;
import com.rokudo.xpense.models.User;

public class DatabaseUtils {
    private static User currentUser;
    public static CollectionReference usersRef = FirebaseFirestore.getInstance()
            .collection("Users");
    public static CollectionReference walletsRef = FirebaseFirestore.getInstance()
            .collection("Wallets");

    public static DocumentReference getYearReference(String walletId, String year) {
        return walletsRef.document(walletId)
                .collection("Statistics")
                .document(year);
    }

    public static CollectionReference getMonthsReference(String walletId, String year) {
        return walletsRef.document(walletId)
                .collection("StatisticsV2")
                .document(year)
                .collection("MonthStatistics");
    }

    public static CollectionReference getTransactionsRef(String walletId) {
        return walletsRef.document(walletId).collection("TransactionsV2");
    }

    public static CollectionReference invitationsRef = FirebaseFirestore.getInstance()
            .collection("Invitations");
    public static CollectionReference bAccountsRef = FirebaseFirestore.getInstance()
            .collection("BAccounts");
    public static StorageReference userPicturesRef = FirebaseStorage.getInstance()
            .getReference("UserPictures");


    public static User getCurrentUser() {
        return currentUser;
    }

    public static void setCurrentUser(User currentUser) {
        DatabaseUtils.currentUser = currentUser;
    }
}

File: app/src/main/java/com/rokudo/xpense/utils/DatabaseUtils.kt
package com.rokudo.xpense.utils

import com.google.firebase.firestore.CollectionReference
import com.google.firebase.firestore.DocumentReference
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.storage.FirebaseStorage
import com.google.firebase.storage.StorageReference
import com.rokudo.xpense.models.User

object DatabaseUtils {
    var currentUser: User? = null

    val usersRef: CollectionReference =
        FirebaseFirestore.getInstance().collection("Users")

    val walletsRef: CollectionReference =
        FirebaseFirestore.getInstance().collection("Wallets")

    val invitationsRef: CollectionReference =
        FirebaseFirestore.getInstance().collection("Invitations")

    val bAccountsRef: CollectionReference =
        FirebaseFirestore.getInstance().collection("BAccounts")

    val userPicturesRef: StorageReference =
        FirebaseStorage.getInstance().getReference("UserPictures")

    fun getYearReference(walletId: String, year: String): DocumentReference {
        require(walletId.isNotEmpty()) { "Wallet ID cannot be null or empty" }
        return walletsRef.document(walletId)
            .collection("Statistics")
            .document(year)
    }

    fun getMonthsReference(walletId: String, year: String): CollectionReference {
        require(walletId.isNotEmpty()) { "Wallet ID cannot be null or empty" }
        return walletsRef.document(walletId)
            .collection("Statistics")
            .document(year)
            .collection("Months")
    }

    fun getTransactionsRef(walletId: String): CollectionReference {
        require(walletId.isNotEmpty()) { "Wallet ID cannot be null or empty" }
        return walletsRef.document(walletId).collection("Transactions")
    }
}

